src/htmx.js

```diff
@@ -73,6 +73,8 @@ return (function () {
                 getCacheBusterParam: false,
                 globalViewTransitions: false,
                 methodsThatUseUrlParams: ["get"],
+                defaultErrorSwapStyle: "none",
+                defaultErrorTarget: "mirror",
```

Comment:
Controversial idea - allow functions as configuration values. Function is evaluated at runtime and the returned value is used as corresponding configuration value. Removes the need for these reserved magic values, but maybe a bit of an overkill for what is essentially a single use-case?

Reply:
I mean, why not adding function on top of that, but I'm not convinced by them **replacing** the magic value.

My goal with those values was to provide an easy, quick & lazy way for the user to setup a global errors swapping strategy, so that it's easy to toggle them. Just set double `mirror` and you'll have errors everywhere, or let `defaultErrorTarget` to mirror then you'll only need to specify `hx-error-swap` and the target will follow `hx-target` everywhere, that kind of thing.

To reproduce the `mirror` logic with a function and without that magic value, you'd have to code yourself the retrieval of standard swap/target strategy (if applicable) in your project, so that doesn't seem to remove the need in my opinion

Also, wouldn't the preferred way for that kind of thing, be to maybe refactor the `htmx:beforeSwap` and add more parameters to it, so that the dev can override the error swap parameters if they want for specific requests, using that even instead of a function global config?

Config is also supposed to be preferably set through meta tags, though maybe is that not relevant for specific use cases where you'd want your users to write some custom JS to set up their own thing anyway?

Reply:
Very valid points, I especially forgot about meta tags, don't think there is a good way to mix them with functions.